web/js/codeworld_shared.js

```diff
@@ -532,13 +532,36 @@ function registerStandardHints(successFunc) {
       found[2] = [];
     }
 
-    const options = found[0].concat(found[1]).concat(found[2]);
+    const mappedTerms = {
+      square: 'rectangle',
+      triangle: 'polygon',
+    };
+
+    for (const [lookedUpTerm, substitute] of Object.entries(mappedTerms)) {
+      if (lookedUpTerm.startsWith(term)) {
```

Comment:
You seem to be disabling fuzzy matching for substitutions here.  Any reason?  I can definitely see students misspelling square or triangle.  I think if you just leave out the `if` statement, the fuzzy scoring will already do the right thing.  It already scores prefix matches much higher than others.

Reply:
No particular reason. I didn't realize that I was disabling fuzzy matching logic, and we should certainly keep it. Thanks for pointing that out!